src/services/activity_generator.rs

```diff
@@ -15,7 +18,9 @@ pub fn generate_budget_goal_activities(calendar: &Calendar, goals: &Vec<Goal>) -
     dbg!(&goals);
     let mut activities: Vec<Activity> = Vec::with_capacity(goals.capacity());
     for goal in goals {
-        let mut goal_activities = Activity::get_activities_from_budget_goal(goal, calendar);
```

Comment:
Why are you changing the generation of Activities from Budget Goals?  
This test case doesn't touch budget Goals at all.

Reply:
You are right, I will return it back in the next commits. But once we face cases which goals have childs and budgets, we need to provide enhancement here.

Reply:
If we're talking long-term solution, just passing the parent won't cut it. There could be multiple levels of children.